### PR TITLE
update p5.ts to 1.5

### PR DESCRIPTION
**Pull request by asukaminato0721**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by wuyudi** [comment]
These Errors are expected :)

=== ERRORS ===



Error in p5
Error: /home/vsts/work/1/s/types/p5/constants.d.ts:69:17
ERROR: 69:17   expect                TypeScript@5.0 compile error: 
Cannot find name 'VIDEO'.
ERROR: 69:25   expect                TypeScript@5.0 compile error: 
Cannot find name 'AUDIO'. Did you mean 'AudioIn'?

/home/vsts/work/1/s/types/p5/lib/addons/p5.sound.d.ts:2658:5
ERROR: 2658:5  no-unnecessary-class  Every member of this class is a constructor. Use functions instead.

/home/vsts/work/1/s/types/p5/test/index.ts:36:45
ERROR: 36:45   semicolon             Missing semicolon

/home/vsts/work/1/s/types/p5/test/test-global/p5-tests.ts:5:3
ERROR: 5:3     expect                TypeScript@5.0 compile error: 
Cannot find name 'createCanvas'.
ERROR: 9:3     expect                TypeScript@5.0 compile error: 
Cannot find name 'background'.
ERROR: 10:3    expect                TypeScript@5.0 compile error: 
Cannot find name 'fill'.
ERROR: 11:3    expect                TypeScript@5.0 compile error: 
Cannot find name 'ellipse'.

**Comment by wuyudi** [comment]
some notifications:

I remove the test for global mode. (But they work.)

I replace `type TYPE= AUDIO | MEDIA` with `type TYPE = "AUDIO" | "MEDIA"` (here seems to have some problems with the origin yui doc)

**Comment by wuyudi** [comment]
convert them to be like the origin type.

File: types/p5/tslint.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
     "extends": "@definitelytyped/dtslint/dt.json",
     "rules": {
         "file-name-casing": false,
-        "unified-signatures": false
+        "unified-signatures": false,
+        "no-unnecessary-class": false
```
Comment (sheetalkamat) [inline types/p5/tslint.json:6]: Please remove this additional disabling of the rule and fix error or disable error on the line where it needs to be

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/p5/tslint.json (hunk 1 above)
Comment (wuyudi) [reply types/p5/tslint.json]: fixed

**Review by sheetalkamat — APPROVED** [review]

**Comment by wuyudi** [comment]
Ready to merge

**Pull request merged**